source/digits_hits/src/GateGlobalActorAdder.cc

```diff
@@ -0,0 +1,80 @@
+/**
+ *  @copyright Copyright 2019 The J-PET Gate Authors. All rights reserved.
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  @file GateGlobalActorAdder.cc
+ */
+
+#include "GateGlobalActorAdder.hh"
+#include "TMath.h"
+#include <iostream>
+
+GateGlobalActorAdder::GateGlobalActorAdder() {}
+
+GateGlobalActorAdder::~GateGlobalActorAdder() {}
+
+void GateGlobalActorAdder::processHit( const GateGlobalActorHit& hit )
+{
+ if ( hit.getParentID() != 0 ) { return; }
```

Comment:
what does it mean?
Why parentID !=0?

Reply:
Because  I want to be sure that we only accept data from particles emitted from source. Example: gamma is emitted from source ( parentID=0), then scatter. During scattering energy deposition is transform to electron (parentID !=0).

I added comment to code about this.

Reply:
What about the case when the gamma undergoes single Compton in one detector then reacts in the second layer and you have multiple scattering then? 
Shouldn't you also merge that cases?

Also, shouldn't you also merge energies from electrons (I think Daria does it)?

Reply:
It doesn't matter.
We have this scenario:
1. gamma scatter once in scintillator A
2. then scatter multiply in scintillator B
As a results adder will generate two hits: hit 1 in scitillator A and 1 hit in scintillator B ( merged hits ). In both cases we have the the same gamma ( parentID=0 and for example trackID=1).

About electron: 
I don't think so - because it generate a lot of problem:
1. in this case how we handle a post polarization and momentum ?
2. now let's think about this situation:  gamma scatters with 100 keV energy deposition and this is a transferred kinetic energy to an electron. This leads us to situation when total deposited energy is 200 keV right? but this energy deposition by electron is caused by msc (multiple scattering).
I think we should only take this under consideration when we use optical photons mechanism -  but we didn't so it not necessary.

In my opinion, let's GateGlobalActorAdder be a tool which only take under consideration an energy deposition from gamma. If we want to use more complex mechanism ( with including optical photons ) it is better to create new adder dedicated for this e.g. GateGlobalActorAdderOpticalPhotons.

Reply:
Ad 1. You assign it from the first Compton scattering.
Ad 2. I dont understand what do you mean. 
Or you calculate the Edep from the multiple scattering of electron, or you calculate it as a difference between Egamma incoming - Egamma outgoing. If your code does the second thing. Then it is fine and you don't need to take into account the electron scattering.

For sure you dont want to take into account optical photons, but it has nothing to do.
So I assume you are calculating it from the gamma energy difference and it is fine